### ISSUE TITLE
Support polling messages with functional binding

### ISSUE THREAD

**Issue by Laures** [spring-cloud/spring-cloud-stream#1956]
Currently it is not possible to have a polling consumer with functional binding active.

spring-cloud-stream deactivates functional binding as soon as the `@EnableBinding` annotation is used (see `FunctionBindingRegistrar`). This annotation i necessary to poll messages.

So while it is possible to have multiple functions as processors/consumers in the same application, there is no way to poll messages in a functional setting. As soon as anything needs to be polled, everything has to be configured with annotations.

**Comment by davidwadden** [comment]
I'm also looking for this capability (original comment [here](https://github.com/spring-cloud/spring-cloud-stream-binder-kafka/issues/882#issuecomment-626355979)).  Will the ability to evolve from `@EnableBinding`-configured streams to Functional configuration also be supported with this forthcoming change?

Thanks
David

> @sobychacko I'm running into this one too trying to evolve an @EnableBinding-configured application to the functional bean declarations (since apparently @StreamListener is set to be deprecated). I'd ideally like to leverage functional definitions alongside the existing channel definitions via @EnableBinding. I'm seeing the the same in my log at startup (traced to FunctionConfiguration.java:665)

> 11:25:50.145 [Test worker] INFO  o.s.c.s.f.FunctionConfiguration$FunctionBindingRegistrar - Functional binding is disabled due to the presense of @EnableBinding annotation in your configuration
Is there any way to use SCSt functional processors alongside @EnableBinding ? I've also got the challenge of adapting from an inbound REST Controller to publishing using the Supplier interface to make a Source of some sort. The examples use a BlockingQueue and the supplier polls the queue, is this going to get polled using the 1Hz defaultPoller (I believe?) ?

**Comment by ChiragMoradiya** [comment]
I encountered this issue while using Spring Cloud Bus with my project. As soon as spring cloud bus was activated, my functional bindings were stopped working. 

Luckily, there was good logging. I found following in my logs:
`onConfiguration$FunctionBindingRegistrar : Functional binding is disabled due to the presense of @EnableBinding annotation in your configuration` 

For now, I have disabled cloud-bus in my project. It is needed to update cloud-config in real-time when the configuration is changed from the Git Repository.
